### PR TITLE
feat: adds a new marker accessibility sample

### LINKED ISSUE #637: New sample for marker accessibility
Now that JSAPI supports accessible markers, we need a new sample to show how this works. This sample should show how to set up markers with a click listener and `title` text, to demonstrate the behavior of these elements for keyboard navigation, screen readers, etc. Here is a draft JSFiddle for what the sample should look like: https://jsfiddle.net/dubnewt/k9qt3ecu/20/

### PR DESCRIPTION
**Pull request by willum070**
This PR adds a new sample to demonstrate how to make markers accessible.

Fixes #637 🦕

### CODE REVIEW

File: samples/marker-accessibility/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  marker.setLabel(labelText);
+
+  // Create an info window using the same title text.
+  const infowindow = new google.maps.InfoWindow({
+    content: titleText,
+  });
+  
+  // Set a click listener to make the marker accessible.
+  marker.addListener("click", () => {
+    infowindow.open(marker.get("map"), marker);
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:77]:
```suggestion
    infowindow.open(marker.getMap(), marker);
```

File: samples/marker-accessibility/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    "Second marker: Airport Mesa",
+    "Third marker: Chapel of the Holy Cross",
+    "Fourth marker: Red Rock Crossing",
+    "Marker five: Bell Rock, the final stop on our tour!"
+  ];
+
+  for (let i = 0; i < titleText.length; ++i) {
+    const pos = tourStops[i];
+    const marker = new google.maps.Marker({
+      position: { lat: pos[0], lng: pos[1],
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:54]: why not pass the title and label to the constructor?

File: samples/marker-accessibility/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+// Adds a click listener and title text to each marker.
+function attachText(marker: google.maps.Marker, titleText: string, labelText: string) {
+  // Set the marker title to provide accessible text.
+  marker.setTitle(titleText);
+  
+  // Add a numeric label to the marker.
+  marker.setLabel(labelText);
+
+  // Create an info window using the same title text.
+  const infowindow = new google.maps.InfoWindow({
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:71]: might be better to use a single infowindow that we update?

File: samples/marker-accessibility/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    content: titleText,
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:72]:
```suggestion
    content: marker.getTitle(),
```

**Review by jpoehnelt — CHANGES_REQUESTED** [review]

File: samples/marker-accessibility/src/index.ts [hunk 5]
```diff
@@ -53,31 +53,22 @@ function initMap(): void {
     const marker = new google.maps.Marker({
       position: { lat: pos[0], lng: pos[1] },
       map: map,
+      title: titleText[i],
+      label: (i + 1).toString(),
     });
-    attachText(marker, titleText[i], (i + 1).toString());
+    setupMarker(marker);
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:59]:
```suggestion
    marker.addListener("click", () => {
      infowindow.close();
      infowindow.setContent(marker.getTitle());
      infowindow.open(marker.getMap(), marker);
    });
```

File: samples/marker-accessibility/src/index.ts [hunk 6]
```diff
@@ -53,31 +53,22 @@ function initMap(): void {
...
-  // Add a numeric label to the marker.
-  marker.setLabel(labelText);
-
-  // Create an info window using the same title text.
-  const infowindow = new google.maps.InfoWindow({
-    content: titleText,
-  });
+function setupMarker(marker: google.maps.Marker) {
+  // Create an info window using the marker's title text.
+  const infowindow = new google.maps.InfoWindow();
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:66]: move this after `const map = google.maps.Map`

File: samples/marker-accessibility/src/index.ts [hunk 7]
```diff
@@ -53,31 +53,22 @@ function initMap(): void {
...
-  marker.setTitle(titleText);
-
-  // Add a numeric label to the marker.
-  marker.setLabel(labelText);
-
-  // Create an info window using the same title text.
-  const infowindow = new google.maps.InfoWindow({
-    content: titleText,
-  });
+function setupMarker(marker: google.maps.Marker) {
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:64]: delete this function

**Review by jpoehnelt — CHANGES_REQUESTED** [review]
This should allow a single infowindow that moves as the user triggers the click event on the marker.

File: samples/marker-accessibility/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  // Create an info window to share between markers.
+  const infowindow = new google.maps.InfoWindow();
+
+  for (let i = 0; i < titleText.length; ++i) {
+    const pos = tourStops[i];
+    const marker = new google.maps.Marker({
+      position: { lat: pos[0], lng: pos[1] },
+      map: map,
+      title: titleText[i],
+      label: (i + 1).toString(),
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:60]:
```suggestion
      label: `${i + 1}`,
```

File: samples/marker-accessibility/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    "Second marker: Airport Mesa",
+    "Third marker: Chapel of the Holy Cross",
+    "Fourth marker: Red Rock Crossing",
+    "Marker five: Bell Rock, the final stop on our tour!",
+  ];
+
+  // Create an info window to share between markers.
+  const infowindow = new google.maps.InfoWindow();
+
+  for (let i = 0; i < titleText.length; ++i) {
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:54]:
```suggestion
titleText.forEach((title, i) => {
```

File: samples/marker-accessibility/src/index.ts [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    });
+
+    // Add a click listener for each marker, and set up the info window.
+    marker.addListener("click", () => {
+      infowindow.close();
+      infowindow.setContent(marker.getTitle());
+      infowindow.open(marker.getMap(), marker);
+    });
+  }
+}
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:70]:
```suggestion
});
```

File: samples/marker-accessibility/src/index.ts [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+  // Create an info window to share between markers.
+  const infowindow = new google.maps.InfoWindow();
+
+  for (let i = 0; i < titleText.length; ++i) {
+    const pos = tourStops[i];
+    const marker = new google.maps.Marker({
+      position: { lat: pos[0], lng: pos[1] },
+      map: map,
+      title: titleText[i],
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:59]:
```suggestion
      title,
```

File: samples/marker-accessibility/src/index.ts [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+  ];
+
+  // Create an info window to share between markers.
+  const infowindow = new google.maps.InfoWindow();
+
+  for (let i = 0; i < titleText.length; ++i) {
+    const pos = tourStops[i];
+    const marker = new google.maps.Marker({
+      position: { lat: pos[0], lng: pos[1] },
+      map: map,
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:58]:
```suggestion
      map,
```

**Review by jpoehnelt — CHANGES_REQUESTED** [review]
simplifying with forEach

could also do `const markers = titleText.map` and return the marker.

File: samples/marker-accessibility/src/index.ts [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+    document.getElementById("map") as HTMLElement,
+    {
+      zoom: 12,
+      center: { lat: 34.84555, lng: -111.8035 },
+    }
+  );
+
+  // LatLng data for the markers.
+  const tourStops: [number, number][] = [
+    [34.8791806, -111.8265049], // Boynton Pass
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:33]: convert these to LatLngLiterals, `{lat: 0, lng: 1}`

File: samples/marker-accessibility/src/index.ts [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+  ];
+
+  // Create an info window to share between markers.
+  const infowindow = new google.maps.InfoWindow();
+
+  // Create the markers.
+  titleText.forEach((title, i) => {
+    const pos = tourStops[i];
+    const marker = new google.maps.Marker({
+      position: { lat: pos[0], lng: pos[1],
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:58]:
```suggestion
      position
```

File: samples/marker-accessibility/src/index.ts [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+    "Fourth marker: Red Rock Crossing",
+    "Marker five: Bell Rock, the final stop on our tour!",
+  ];
+
+  // Create an info window to share between markers.
+  const infowindow = new google.maps.InfoWindow();
+
+  // Create the markers.
+  titleText.forEach((title, i) => {
+    const pos = tourStops[i];
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:56]:
```suggestion
    const position = tourStops[i];
```

File: samples/marker-accessibility/src/index.ts [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+  const map = new google.maps.Map(
+    document.getElementById("map") as HTMLElement,
+    {
+      zoom: 12,
+      center: { lat: 34.84555, lng: -111.8035 },
+    }
+  );
+
+  // LatLng data for the markers.
+  const tourStops: [number, number][] = [
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:32]:
```suggestion
  const tourStops: google.maps.LatLngLiteral[] = [
```

**Review by jpoehnelt — CHANGES_REQUESTED** [review]

File: samples/marker-accessibility/src/index.ts [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+  // Set title text for the markers.
+  const titleText: string[] = [
+    "The first marker (Boynton Pass) receives the initial focus when tab " +
+      "is pressed. Use arrow keys to move between markers; press tab again to " +
+      "cycle through the map controls.",
+    "Second marker: Airport Mesa",
+    "Third marker: Chapel of the Holy Cross",
+    "Fourth marker: Red Rock Crossing",
+    "Marker five: Bell Rock, the final stop on our tour!",
+  ];
```
Comment (chrisjshull) [inline samples/marker-accessibility/src/index.ts:49]: Can the data structures holding the latLngs and the titles be combined? seems more realistic (and easier) that way?

File: samples/marker-accessibility/src/index.ts [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+
+  // Create an info window to share between markers.
+  const infowindow = new google.maps.InfoWindow();
```
Comment (chrisjshull) [inline samples/marker-accessibility/src/index.ts:52]: s/infowindow/infoWindow/

File: samples/marker-accessibility/src/index.ts [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+
+  // Create the markers.
+  titleText.forEach((title, i) => {
+    const marker = new google.maps.Marker({
+      position: {
+        lat: tourStops[i][0], lng: tourStops[i][1],
+      },
+      map,
+      title,
+      label: `${i + 1}`,
```
Comment (chrisjshull) [inline samples/marker-accessibility/src/index.ts:62]: if there is a title then the label is not presented for a11y
if the number is important, best practice might be to make the title `(i + 1) + '. ' + title`

File: samples/marker-accessibility/src/index.ts [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+    });
+    
+    // Add a click listener for each marker, and set up the info window.
+    marker.addListener("click", () => {
+      infowindow.close();
```
Comment (chrisjshull) [inline samples/marker-accessibility/src/index.ts:67]: might be good to comment why the close is needed

File: samples/marker-accessibility/src/index.ts [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+
+  // Set title text for the markers.
+  const titleText: string[] = [
+    "The first marker (Boynton Pass) receives the initial focus when tab " +
+      "is pressed. Use arrow keys to move between markers; press tab again to " +
+      "cycle through the map controls.",
+    "Second marker: Airport Mesa",
+    "Third marker: Chapel of the Holy Cross",
+    "Fourth marker: Red Rock Crossing",
+    "Marker five: Bell Rock, the final stop on our tour!",
```
Comment (chrisjshull) [inline samples/marker-accessibility/src/index.ts:48]: people don't usually title markers with "marker" in them.

also, there is a lack of parallelism here.

File: samples/marker-accessibility/src/index.ts [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+    [34.832149, -111.7695277], // Chapel of the Holy Cross
+    [34.823736, -111.8001857], // Red Rock Crossing
+    [34.800326, -111.7665047], // Bell Rock
+  ];
+
+  // Set title text for the markers.
+  const titleText: string[] = [
+    "The first marker (Boynton Pass) receives the initial focus when tab " +
+      "is pressed. Use arrow keys to move between markers; press tab again to " +
+      "cycle through the map controls.",
```
Comment (chrisjshull) [inline samples/marker-accessibility/src/index.ts:44]: I don't think we want this meta-description in any one marker. I would make all the marker titling consistent.

File: samples/marker-accessibility/src/index.ts [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+      center: { lat: 34.84555, lng: -111.8035 },
+    }
+  );
+
+  // Set LatLng and title text for the markers. The first marker (Boynton Pass)
+  // receives the initial focus when tab is pressed. Use arrow keys to
+  // move between markers; press tab again to cycle through the map controls.
+  const tourStops: [google.maps.LatLngLiteral, string][] = [
+  	[{lat: 34.8791806, lng: -111.8265049},"Boynton Pass"],
+    [{lat: 34.8559195, lng: -111.7988186}, "Airport Mesa"],
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:36]: `npm run format` or `bazel run //:eslint-fix`

File: samples/marker-accessibility/src/index.ts [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+    [{lat: 34.800326, lng: -111.7665047}, "Bell Rock"],
+	];
+
+  // Create an info window to share between markers.
+  const infoWindow = new google.maps.InfoWindow();
+
+  // Create the markers.
+  tourStops.forEach((item, i) => {
+    const marker = new google.maps.Marker({
+      position: item[0],
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:48]:
```suggestion
      position,
```

File: samples/marker-accessibility/src/index.ts [hunk 25]
```diff
@@ -0,0 +1,64 @@
...
+      map,
+      title: `${i + 1}` + ". " + item[1],
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:50]:
```suggestion
      title: `${i + 1}.${title}`,
```

**Review by jpoehnelt — CHANGES_REQUESTED** [review]

File: samples/marker-accessibility/src/index.ts [hunk 26]
```diff
@@ -0,0 +1,64 @@
...
+    [{lat: 34.832149, lng: -111.7695277}, "Chapel of the Holy Cross"],
+    [{lat:34.823736, lng: -111.8001857}, "Red Rock Crossing"],
+    [{lat: 34.800326, lng: -111.7665047}, "Bell Rock"],
+	];
+
+  // Create an info window to share between markers.
+  const infoWindow = new google.maps.InfoWindow();
+
+  // Create the markers.
+  tourStops.forEach((item, i) => {
```
Comment (jpoehnelt) [inline samples/marker-accessibility/src/index.ts:46]:
```suggestion
  tourStops.forEach(([position, title], i) => {
```

**Pull request merged**